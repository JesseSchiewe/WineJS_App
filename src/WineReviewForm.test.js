import React from 'react';
import ReactDOM from 'react-dom';
import WineReviewForm from './WineReviewForm';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

// afterEach(cleanup)

// "Review" Type (Brand new review, not filling in past review data.
jest.mock("react-router-dom", ()  => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:3000/review"
    })
}));

it("renders without crashing - Review Type", () => {
    const div = document.createElement("div");
    ReactDOM.render(<WineReviewForm></WineReviewForm>, div)
});

it("renders page correctly", () => {
    const {getByTestId} = render(<WineReviewForm></WineReviewForm>)
    expect(getByTestId('FullReview')).toHaveTextContent("Nose Intensity")
});

it("matches snapshot", () => {
    const tree = renderer.create(<WineReviewForm/>).toJSON();
    expect(tree).toMatchSnapshot();
})


// "ReviewResult" Type (Load existing data into form)
// cleanup
jest.mock("react-router-dom", ()  => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:3000/reviewresult"
    })
}));

it("renders without crashing - Review Type", () => {
    const div = document.createElement("div2");
    ReactDOM.render(<WineReviewForm></WineReviewForm>, div)
});

it("renders page correctly", () => {
    const {getByTestId} = render(<WineReviewForm></WineReviewForm>)
    expect(getByTestId('FullReview')).toHaveTextContent("Balance")
});

it("matches snapshot 2", () => {
    const tree = renderer.create(<WineReviewForm/>).toJSON();
    expect(tree).toMatchSnapshot();
})
