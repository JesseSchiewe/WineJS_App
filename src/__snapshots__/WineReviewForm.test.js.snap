// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="FullReview"
>
  <div
    className="FavoriteWines"
    hidden={false}
  >
    <form />
    <p />
  </div>
  <div
    className="ReviewPage"
  >
    <form
      className="reviewform"
      onSubmit={[Function]}
    >
      <div
        className="formbackground"
      >
        <div>
          This is a past review. You may edit the review data and save the changes or delete the review from this page.
          <h2>
            Review Date
          </h2>
        </div>
        <input
          defaultValue="9-21-2021"
          id="ReviewDate"
          name="ReviewDate"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
        <h2>
          Producer
        </h2>
        <input
          name="Producer"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Who makes the wine?"
          type="text"
        />
        <h2>
          Appellation
        </h2>
        <input
          name="Appellation"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Where is the wine from? Ex: Napa Valley"
          type="text"
        />
        <h2>
          Wine Name
        </h2>
        <input
          name="WineName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Wine Name"
          type="text"
        />
        <h2>
          Vintage
        </h2>
        <input
          max="2030"
          min="1900"
          name="Vintage"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Year"
          type="number"
        />
        <h3>
          Nose Intensity
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="NoseIntensity"
          max="5"
          min="1"
          name="NoseIntensity"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="NoseIntensityNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <div
          hidden={true}
          name="AromaSelector"
        >
          <input
            id="Aromas"
            name="Aromas"
            onBlur={[Function]}
            onChange={[Function]}
            type="hidden"
          />
          <div
            className="selectedAromas"
          >
            Selected Aromas: 
          </div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-5duvv3-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Aroma Selector
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-6-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="AromaSelector"
                type="hidden"
              />
            </div>
          </div>
        </div>
        <h3>
          Flavor Intensity
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="FlavorIntensity"
          max="10"
          min="1"
          name="FlavorIntensity"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="FlavorIntensityNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <h3>
          Flavor Characteristics
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="FlavorCharacteristics"
          max="25"
          min="1"
          name="FlavorCharacteristics"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="FlavorCharacteristicsNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <div
          hidden={true}
          name="FlavorSelector"
        >
          <input
            id="Flavors"
            name="Flavors"
            onBlur={[Function]}
            onChange={[Function]}
            type="hidden"
          />
          <div
            className="selectedFlavors"
          >
            Selected Flavors: 
          </div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-5duvv3-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Flavor Selector
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-7-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="FlavorSelector"
                type="hidden"
              />
            </div>
          </div>
        </div>
        <h3>
          Balance
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="Balance"
          max="5"
          min="1"
          name="Balance"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="BalanceNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <h3>
          Length
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="Length"
          max="5"
          min="1"
          name="Length"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="LengthNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <h4>
          Total
        </h4>
        <div
          className="value"
        >
          NaN
        </div>
        <input
          className="hidethis"
          id="Total"
          name="Total"
          onBlur={[Function]}
          onChange={[Function]}
          type="hidden"
          value="NaN"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Purchase Info
        </button>
        <h2>
          
        </h2>
        <input
          hideit="true"
          name="ActualPrice"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$"
          type="number"
        />
        <h2>
          
        </h2>
        <input
          hideit="true"
          name="WineValue"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$"
          type="number"
        />
        <div
          className="WineTools"
        >
          <button
            className="winetoolsbutton"
            onClick={[Function]}
            type="button"
          >
            Wine Colors
          </button>
        </div>
        <div
          hidden={true}
        >
          <img
            alt="Wine Color Chart"
            src="WineColorChart.jpg"
            width={400}
          />
        </div>
        <div
          className="WineTools"
        >
          <button
            className="winetoolsbutton"
            onClick={[Function]}
            type="button"
          >
            Wine Tasting Grid
          </button>
        </div>
        <div
          hidden={true}
        >
          <img
            alt="Wine Tasting Grid"
            src="WineTastingGrid.jpg"
            width={400}
          />
        </div>
        <h2>
          Tasting Notes
        </h2>
        <textarea
          name="TastingNotes"
          onBlur={[Function]}
          onChange={[Function]}
        />
        <div>
          <input
            defaultValue="Update"
            onClick={[Function]}
            type="submit"
          />
          <p />
          <input
            defaultValue="Delete Review"
            name="DeleteReview"
            onClick={[Function]}
            type="delete"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="FullReview"
>
  <div
    className="FavoriteWines"
    hidden={false}
  >
    <form />
    <p />
  </div>
  <div
    className="ReviewPage"
  >
    <form
      className="reviewform"
      onSubmit={[Function]}
    >
      <div
        className="formbackground"
      >
        <div>
          This is a past review. You may edit the review data and save the changes or delete the review from this page.
          <h2>
            Review Date
          </h2>
        </div>
        <input
          defaultValue="9-21-2021"
          id="ReviewDate"
          name="ReviewDate"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
        <h2>
          Producer
        </h2>
        <input
          name="Producer"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Who makes the wine?"
          type="text"
        />
        <h2>
          Appellation
        </h2>
        <input
          name="Appellation"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Where is the wine from? Ex: Napa Valley"
          type="text"
        />
        <h2>
          Wine Name
        </h2>
        <input
          name="WineName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Wine Name"
          type="text"
        />
        <h2>
          Vintage
        </h2>
        <input
          max="2030"
          min="1900"
          name="Vintage"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Year"
          type="number"
        />
        <h3>
          Nose Intensity
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="NoseIntensity"
          max="5"
          min="1"
          name="NoseIntensity"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="NoseIntensityNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <div
          hidden={true}
          name="AromaSelector"
        >
          <input
            id="Aromas"
            name="Aromas"
            onBlur={[Function]}
            onChange={[Function]}
            type="hidden"
          />
          <div
            className="selectedAromas"
          >
            Selected Aromas: 
          </div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-5duvv3-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Aroma Selector
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-12-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="AromaSelector"
                type="hidden"
              />
            </div>
          </div>
        </div>
        <h3>
          Flavor Intensity
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="FlavorIntensity"
          max="10"
          min="1"
          name="FlavorIntensity"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="FlavorIntensityNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <h3>
          Flavor Characteristics
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="FlavorCharacteristics"
          max="25"
          min="1"
          name="FlavorCharacteristics"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="FlavorCharacteristicsNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <div
          hidden={true}
          name="FlavorSelector"
        >
          <input
            id="Flavors"
            name="Flavors"
            onBlur={[Function]}
            onChange={[Function]}
            type="hidden"
          />
          <div
            className="selectedFlavors"
          >
            Selected Flavors: 
          </div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-5duvv3-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Flavor Selector
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-13-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="FlavorSelector"
                type="hidden"
              />
            </div>
          </div>
        </div>
        <h3>
          Balance
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="Balance"
          max="5"
          min="1"
          name="Balance"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="BalanceNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <h3>
          Length
          <button
            className="infobutton"
            onClick={[Function]}
            type="button"
          >
            info
          </button>
          <div>
            <h5>
              
            </h5>
          </div>
        </h3>
        <div
          className="value"
        />
        <input
          defaultValue="0"
          id="Length"
          max="5"
          min="1"
          name="Length"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Show/hide notes
        </button>
        <textarea
          hideit="true"
          name="LengthNotes"
          onBlur={[Function]}
          onChange={[Function]}
          type="small"
        />
        <h4>
          Total
        </h4>
        <div
          className="value"
        >
          NaN
        </div>
        <input
          className="hidethis"
          id="Total"
          name="Total"
          onBlur={[Function]}
          onChange={[Function]}
          type="hidden"
          value="NaN"
        />
        <button
          onClick={[Function]}
          type="button"
          value=""
        >
          Purchase Info
        </button>
        <h2>
          
        </h2>
        <input
          hideit="true"
          name="ActualPrice"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$"
          type="number"
        />
        <h2>
          
        </h2>
        <input
          hideit="true"
          name="WineValue"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$"
          type="number"
        />
        <div
          className="WineTools"
        >
          <button
            className="winetoolsbutton"
            onClick={[Function]}
            type="button"
          >
            Wine Colors
          </button>
        </div>
        <div
          hidden={true}
        >
          <img
            alt="Wine Color Chart"
            src="WineColorChart.jpg"
            width={400}
          />
        </div>
        <div
          className="WineTools"
        >
          <button
            className="winetoolsbutton"
            onClick={[Function]}
            type="button"
          >
            Wine Tasting Grid
          </button>
        </div>
        <div
          hidden={true}
        >
          <img
            alt="Wine Tasting Grid"
            src="WineTastingGrid.jpg"
            width={400}
          />
        </div>
        <h2>
          Tasting Notes
        </h2>
        <textarea
          name="TastingNotes"
          onBlur={[Function]}
          onChange={[Function]}
        />
        <div>
          <input
            defaultValue="Update"
            onClick={[Function]}
            type="submit"
          />
          <p />
          <input
            defaultValue="Delete Review"
            name="DeleteReview"
            onClick={[Function]}
            type="delete"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
